### PR TITLE
Adding begin/end to build definitions

### PR DESCRIPTION
**Pull request by JohnTortugo**
Continuation of this PR: https://github.com/dotnet/corefx/pull/28624

Related to this issue: https://github.com/dotnet/core-eng/issues/2964

### CODE REVIEW

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 1]
```diff
@@ -1,5 +1,27 @@
 {
   "build": [
+    {
+      "environment": {},
+      "enabled": true,
+      "continueOnError": true,
+      "displayName": "run begin.sh",
+      "timeoutInMinutes": 0,
+      "condition": "always()",
+      "refName": "TaskRunBegin",
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: Probably worth just excluding refNames as they get generated automatically.

File: buildpipeline/DotNet-CoreFx-Trusted-Windows-NoTest.json [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+      "task": {
+        "id": "e213ff0f-5d5c-4791-802d-52ea3e7be1f1",
+        "versionSpec": "2.*",
+        "definitionType": "task"
+      },
+      "inputs": {
+        "targetType": "inline",
+        "filePath": "",
+        "arguments": "",
+        "script": "if (Test-Path \"$(AgentToolsPath)\\begin.ps1\") {\n   \"begin.ps1 script found. Executing...\"\n   cd $(AgentToolsPath)\n   & $(AgentToolsPath)\\begin.ps1\n} else {\n   \"begin.ps1 script does not exist. Moving on...\"\n}",
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreFx-Trusted-Windows-NoTest.json]: Inconsistency with begin/end.sh where here you echo found script and executing switching directories. I think it is a good idea to echo executing script with the location to the script and probably avoid switching directories.

Comment (JohnTortugo) [reply buildpipeline/DotNet-CoreFx-Trusted-Windows-NoTest.json]: Thanks. Done.

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json (hunk 1 above)
Comment (JohnTortugo) [reply buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: Thanks. Done.

**Pull request merged**